runtime/src/accounts_hash.rs

```diff
@@ -869,37 +821,125 @@ impl AccountsHasher {
         min_index: usize,
         bin: usize,
         first_items: &'a mut Vec<Pubkey>,
-        pubkey_division: &'b [SortedDataByPubkey<'b>],
-        indexes: &'a mut [usize],
+        sorted_data_by_pubkey: &'a [&'b [CalculateHashIntermediate]],
```

Comment:
yes, references can be cleaned up in the whole call chain. This has been true since we went to a file based intermediate data transfer instead of keeping refs to slices in memory.